rsfbclient-rust/src/wire.rs

```diff
@@ -267,19 +278,25 @@ pub fn allocate_statement(db_handle: u32) -> Bytes {
 
 /// Prepare statement request. Use u32::MAX as `stmt_handle` if the statement was allocated
 /// in the previous request
-pub fn prepare_statement(tr_handle: u32, stmt_handle: u32, dialect: u32, query: &str) -> Bytes {
+pub fn prepare_statement(
+    tr_handle: u32,
+    stmt_handle: u32,
+    dialect: u32,
+    query: &str,
+    charset: &Charset,
+) -> Result<Bytes, FbError> {
     let mut req = BytesMut::with_capacity(28 + query.len() + XSQLDA_DESCRIBE_VARS.len());
```

Comment:
The capacity calculation is counting always the `UTF-8` byte count, so it may be wrong in other encodings.

Reply:
Fixed!